components/script/script_thread.rs

```diff
@@ -1126,6 +1148,9 @@ impl ScriptThread {
         // Store new resizes, and gather all other events.
         let mut sequential = vec![];
 
+        // Notify the background-hang-monitor we are waiting for an event.
+        self.background_hang_monitor.notify_wait();
```

Comment:
Let's add a call to `notify_activity_to_hang_monitor` inside of `ScriptThread::profile_event`, which will take care of a bunch of cases below this where we are actually not waiting.